File: civitasruby/pkg/Civitas-0.0.1/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class Dado
   include Singleton
   
  attr_reader :ultimoResultado
  def initialize
    @ultimoResultado=1
    @debug=false
    @@SalidaCarcel=5
  end
  
  def tirar
    if(!@debug)
      @ultimoResultado=1+rand(6).to_i 
    else
      @ultimoResultado=1
    end
    
    return @ultimoResultado
  end
  
  def salgo_de_la_carcel
    a=tirar
    return a>=5
  end
  
  def quien_empieza(n)
      
      numero=rand(n).to_i
      
    return numero
   
  end
  
  def set_debug(d)
    if(d)
      Diario.instance.ocurre_evento('Debug on(avanza de unidad en unidad)')
      @debug=d
    else
      @debug=d
       Diario.instance.ocurre_evento('Debug off(avanza segun dado')
    end
  end
  
  
  
  end

end

File: JavaCivitas/src/civitas/Dado.java

package civitas;

import java.util.Random;

/**
 *
 * @author Daniel
 * 
 */
public class Dado {
    
    // Atributos de instancia
    private Random random = new Random();
    private int ultimoResultado;
    private boolean debug;
    
    //Atributos de clase
    static final private Dado instance = new Dado();
    private final int SalidaCarcel = 5;
    
    private Dado(){
        debug = false;
        
        ultimoResultado = 0;
    }
    
    public static Dado getInstance(){
        return instance;
    }
    
    int tirar (){
        Random r1 = new Random ();
        
        ultimoResultado = 1;
        if (!debug)
            ultimoResultado = r1.nextInt(7);
        
        return ultimoResultado;
    }
    
    boolean salgoDeLaCarcel (){
        boolean salgo = false;
        
        if (tirar() >= 5)
                salgo = true;
        
        return salgo;
    }
    
    int quienEmpieza (int n){
        Random r = new Random();    
        int este_sale = r.nextInt(n) + 1;
        return este_sale;
    }
    
    public void setDebug (Boolean d){
        
        if (d){
            debug = true;
            Diario.getInstance().ocurreEvento("\n[DIARIO] Debug esta activado");
        }
        else {
            debug = false;
            Diario.getInstance().ocurreEvento("\n[DIARIO] Debug esta desactivado");

        }
        
    }
    int getUltimoResultado(){
        return ultimoResultado;
    }
    
    
}
